src/webui/src/App.tsx

```diff
@@ -33,8 +39,34 @@ class App extends React.Component<{}, AppState> {
         this.setState(state => ({ trialsUpdateBroadcast: state.trialsUpdateBroadcast + 1 }));
         this.timerId = window.setTimeout(this.refresh, this.state.interval * 1000);
         this.setState({ metricGraphMode: (EXPERIMENT.optimizeMode === 'minimize' ? 'min' : 'max') });
+        // final result is legal
+        // get a succeed trial，see final result data's format
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        this.dataFormatimer = window.setInterval(this.getFinalDataFormat, this.state.interval * 1000);
     }
 
+    getFinalDataFormat = (): void => {
+        for(let i = 0; this.state.isilLegalFinal === false; i++){
+            if(TRIALS.succeededTrials()[0] !== undefined && TRIALS.succeededTrials()[0].final !== undefined){
```

Comment:
What if we don't have succeeded trials?

Reply:
line 64, quit for loop.

Reply:
I mean `TRIALS.succeddedTrials()` will return an empty array, right? What happens to `x[0]` if `x` is an empty array?